java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -462,7 +463,27 @@ private void executeMethodInvocation(MethodInvocationTree mit) {
     ProgramState.Pop unstack = programState.unstackValue(mit.arguments().size() + 1);
     programState = unstack.state;
     logState(mit);
-    programState = programState.stackValue(constraintManager.createMethodSymbolicValue(mit, unstack.values));
+    final SymbolicValue resultValue = constraintManager.createMethodSymbolicValue(mit, unstack.values);
+    programState = programState.stackValue(resultValue);
+    if (isNonNullMethod(mit)) {
+      programState = programState.addConstraint(resultValue, ObjectConstraint.NOT_NULL);
+    }
+  }
+
+  private boolean isNonNullMethod(MethodInvocationTree mit) {
+    IdentifierTree identifier;
```

Comment:
This whole to get the identifier is useless. You can directly use `mit.symbol()` to get the symbol of the method.